### PR TITLE
Add clustering doc

### LINKED ISSUE #145: Clustering doc
#### OpenStack Release
Liberty +

#### Description
Add new documentation to support clustering; should use new template that includes prereqs, caveats, & config instructions.

### PR DESCRIPTION
**Pull request by jputrino**
@jlongstaf @szakeri 

#### What issues does this address?
Fixes #145 
WIP #138, #137 

#### What's this change do?
- adds new clustering doc
- removes references to auto-sync
- adds new multi-tenancy doc
- updates cert manager doc 
- adds diagram for cert manager
- adds glossary
- adds/implements new template for documenting features

#### Where should the reviewer start?

To build the documentation locally:
- clone my fork (`git clone -b feature-clustering https://github.com/jputrino/f5-openstack-lbaasv2-driver.git`)
- `pip install requirements.docs.txt`
- `sphinx-build docs/ docs/_build`

### CODE REVIEW

**Comment by jputrino** [comment]
@pjbreaux - Adding you for review of content.

**Comment by jputrino** [comment]
@szakeri @pjbreaux -- I added my fork of this project to read the docs so you can review the documentation without having to build it locally.

http://lbaasv2-test.readthedocs.io/en/feature-clustering/includes/topic_clustering.html

File: docs/includes/topic_cert-manager.rst [hunk 1]
```diff
@@ -1,12 +1,64 @@
...
+
+
+Configuration
+-------------
+
+1. Edit the :ref:`Agent Configuration File`:
+
+.. code-block:: text
+
+    $ sudo emacs /etc/neutron/services/f5/f5-openstack-agent.ini
```
Comment (szakeri) [inline docs/includes/topic_cert-manager.rst]: In the LbaaSv2 and F5 Agent docs, vi is used as editor, I was wondering if emacs is used here for any specific reason?

File: docs/includes/topic_clustering.rst [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    # pair - active/standby two device HA
+    # scalen - active device cluster
+    #
+    #
+    f5_ha_type = pair
+    #
+    #
+
+
+.. Further Reading
```
Comment (szakeri) [inline docs/includes/topic_clustering.rst]: Also I think we need to update the icontrol_hostname field in the Agent ini file. Its value will be a list of comma separated VEs floating IP addresses.

File: docs/includes/topic_cert-manager.rst (hunk 1 above)
Comment (jputrino) [inline docs/includes/topic_cert-manager.rst]: None, other than that I just like emacs better. ;)

File: docs/includes/topic_clustering.rst (hunk 2 above)
Comment (jputrino) [inline docs/includes/topic_clustering.rst]: I think actually making changes to the ini file falls under @jlongstaf's purview, but I do need to know if any changes are made -- I've included sample config files in addition to providing the examples in the feature docs.

Comment (jlongstaf) [inline docs/includes/topic_clustering.rst]: As @szakeri suggested we should document that a comma separated list of IP addresses for icontrol_hostname in the .ini file is needed. Users with a BIG-OIP cluster must define the IP addresses for each BIG-IP in their cluster. For example:

icontrol_hostname = 10.190.7.232, 10.190.7.233

Comment (jputrino) [inline docs/includes/topic_clustering.rst]: Ahh, will do! I misunderstood.

File: docs/_static/f5-openstack-agent.gre.ini [hunk 3]
```diff
@@ -81,13 +81,13 @@ periodic_interval = 10
...
 #
-# HA model
+# HA mode
 #
 # Device can be required to be:
 #
 # standalone - single device no HA
-# pair - active/standby two device HA(Not supported)
-# scalen - active device cluster(Not supported)
+# pair - active/standby two device HA
```
Comment (pjbreaux) [inline docs/_static/f5-openstack-agent.gre.ini]: I know talked to Jeff about this earlier, but is an Active/Active cluster referred as an HA Pair?

Comment (jputrino) [inline docs/_static/f5-openstack-agent.gre.ini]: Yup. Pair refers to active/active or active/standby, as I understand it.

Comment (jputrino) [inline docs/_static/f5-openstack-agent.gre.ini]: The config file should probably be updated to reflect that.

**Comment by jputrino** [comment]
I've got the changes @szakeri requested in the iControl driver settings doc; I'm going to close out this PR and just include the clustering doc in the larger user guide doc PR I expect to open later today.

**Pull request closed without merging**